keymanapp/keyman
# feat(mac): invoking Keyman Core from Keyman Engine for Mac  🍕

Initial steps in integration of Keyman Engine with Core:
1. change to build script to trigger a build of Keyman Core library when building Keyman Engine
2. Keyman Engine includes new wrapper classes that encapsulate the communication with Core
3. Wrapper classes are dependent on Keyman Core headers and library
4. Keyman Engine creates a new wrapper instance whenever the keyboard is changed (and it should be disposing of all resources related to the previous keyboard but this is not yet confirmed)

# TODO

- [ ] Keyman Engine should be disposing of all resources related to the previous keyboard but this is not yet confirmed

# User Testing

- **TEST_AMHARIC**: 
Install and select the Amharic keyboard. Run Pages and verify that

1) typing <kbd>t</kbd><kbd>a</kbd> produces **ታ**
2) typing <kbd>t</kbd>, left-arrow, <kbd>a</kbd> produces **አት**
3) typing <kbd>t</kbd><kbd>t</kbd>, left-arrow, <kbd>a</kbd> produces **ታት**
4) typing <kbd>t</kbd><kbd>t</kbd>, then mouse-clicking between the two **ታ** characters, and then typing <kbd>a</kbd> produces **ታት**

- **TEST_IPA**: 
Switch to the "IPA (SIL)" keyboard. Run Pages, type <kbd>n></kbd> and verify that the result is **ŋ**

- **TEST_KOREAN**: 
Switch to "Korean KORDA Jamo (SIL)" keyboard. Run Pages and type <kbd>han</kbd>, space, <kbd>geul</kbd>, space Verify that the result is "한글"

- **TEST_CONTEXT**: 
Switch to the "IPA (SIL)" keyboard. Run Pages
1) Type <kbd>an</kbd>, enter
2) Type <kbd>m</kbd>, enter
3) click after the letter n
4) type<kbd>></kbd>
and verify that the output is
aŋ
m
